### ISSUE TITLE
[CRASH] <After running a redissearch knn command, redis is down>

### ISSUE THREAD

**Issue by xiguazhiPrince** [RediSearch/RediSearch#3465]
Notice!
- If a Redis module was involved, please open an issue in the module's repo instead!
- If you're using docker on Apple M1, please make sure the image you're using was compiled for ARM!

**Crash report**

Paste the complete crash log between the quotes below. Please include a few lines from the log preceding the crash report to provide some context.

```


=== REDIS BUG REPORT START: Cut & paste starting from here ===
1:M 13 Mar 2023 16:49:56.017 # Redis 6.2.6 crashed by signal: 11, si_code: 1
1:M 13 Mar 2023 16:49:56.017 # Accessing address: (nil)
1:M 13 Mar 2023 16:49:56.017 # Crashed running the instruction at: 0x7fb9da8e936e

------ STACK TRACE ------
EIP:
/usr/lib/redis/modules/redisearch.so(+0x12536e)[0x7fb9da8e936e]

Backtrace:
/lib/x86_64-linux-gnu/libpthread.so.0(+0x14140)[0x7fb9db7a1140]
/usr/lib/redis/modules/redisearch.so(+0x12536e)[0x7fb9da8e936e]
/usr/lib/redis/modules/redisearch.so(computeDistances+0xea)[0x7fb9da8e967a]
/usr/lib/redis/modules/redisearch.so(+0x125c0d)[0x7fb9da8e9c0d]
/usr/lib/redis/modules/redisearch.so(+0x149aba)[0x7fb9da90daba]
/usr/lib/redis/modules/redisearch.so(+0x149da8)[0x7fb9da90dda8]
/usr/lib/redis/modules/redisearch.so(+0x14a3a5)[0x7fb9da90e3a5]
/usr/lib/redis/modules/redisearch.so(+0x149e98)[0x7fb9da90de98]
/usr/lib/redis/modules/redisearch.so(sendChunk+0x92)[0x7fb9da8bf212]
/usr/lib/redis/modules/redisearch.so(AREQ_Execute+0x19)[0x7fb9da8bf809]
/usr/lib/redis/modules/redisearch.so(+0xfba2c)[0x7fb9da8bfa2c]
redis-server *:6379(RedisModuleCommandDispatcher+0x53)[0x55e997a7c453]
redis-server *:6379(call+0xa1)[0x55e9979f11a1]
redis-server *:6379(processCommand+0x593)[0x55e9979f2d23]
redis-server *:6379(processInputBuffer+0xf8)[0x55e997a05ec8]
redis-server *:6379(+0xfbbb8)[0x55e997a9dbb8]
redis-server *:6379(aeProcessEvents+0x292)[0x55e9979e9e32]
redis-server *:6379(aeMain+0x1d)[0x55e9979ea09d]
redis-server *:6379(main+0x316)[0x55e9979e62b6]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xea)[0x7fb9db5eed0a]
redis-server *:6379(_start+0x2a)[0x55e9979e678a]

------ REGISTERS ------
1:M 13 Mar 2023 16:49:56.019 # 
RAX:0000000000000000 RBX:00007fb9db023af0
RCX:00007fb9db023960 RDX:00007fb9db023b90
RDI:00007fb9d7574f80 RSI:00007fb9db023af0
RBP:00007fb9db023960 RSP:00007ffcc43cd8c0
R8 :00007ffcc43cd904 R9 :0000000000000001
R10:0000000000000001 R11:0000000000000008
R12:00007fb9db10db80 R13:0000000000000000
R14:00007ffcc43cd904 R15:00007fb9db10db80
RIP:00007fb9da8e936e EFL:0000000000010202
CSGSFS:002b000000000033
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8cf) -> 00007fb9dab092bc
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8ce) -> 00007ffcc43cd988
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8cd) -> 00007fb9db0239b0
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8cc) -> 00007ffcc43cd988
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8cb) -> 00007fb9db10db80
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8ca) -> 0000000000000006
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c9) -> 00007fb9db023b90
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c8) -> 7f800000db5c28f0
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c7) -> 52b225e6db2009c0
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c6) -> 00007ffcc43cd95f
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c5) -> 00007fb9da8e967a
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c4) -> 00007fb9db023af0
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c3) -> 00007ffcc43cd904
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c2) -> 00007fb9d755d085
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c1) -> 00007fb9db023960
1:M 13 Mar 2023 16:49:56.019 # (00007ffcc43cd8c0) -> 00007ffcc43cd908

------ INFO OUTPUT ------
# Server
redis_version:6.2.6
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:d572e585a9e438fa
redis_mode:standalone
os:Linux 5.10.16.3-microsoft-standard-WSL2 x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:10.2.1
process_id:1
process_supervised:no
run_id:e37023cc8fb5be9bb7c283b2333b836a02a2cfcf
tcp_port:6379
server_time_usec:1678726196017723
uptime_in_seconds:51
uptime_in_days:0
hz:10
configured_hz:10
lru_clock:1004595
executable:/data/redis-server
config_file:
io_threads_active:0

# Clients
connected_clients:4
cluster_connections:0
maxclients:10000
client_recent_max_input_buffer:24
client_recent_max_output_buffer:0
blocked_clients:0
tracking_clients:0
clients_in_timeout_table:0

# Memory
used_memory:2056536
used_memory_human:1.96M
used_memory_rss:17702912
used_memory_rss_human:16.88M
used_memory_peak:2068840
used_memory_peak_human:1.97M
used_memory_peak_perc:99.41%
used_memory_overhead:1002864
used_memory_startup:904728
used_memory_dataset:1053672
used_memory_dataset_perc:91.48%
allocator_allocated:2283192
allocator_active:2674688
allocator_resident:5529600
total_system_memory:2078142464
total_system_memory_human:1.94G
used_memory_lua:37888
used_memory_lua_human:37.00K
used_memory_scripts:0
used_memory_scripts_human:0B
number_of_cached_scripts:0
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
allocator_frag_ratio:1.17
allocator_frag_bytes:391496
allocator_rss_ratio:2.07
allocator_rss_bytes:2854912
rss_overhead_ratio:3.20
rss_overhead_bytes:12173312
mem_fragmentation_ratio:9.01
mem_fragmentation_bytes:15737704
mem_not_counted_for_evict:0
mem_replication_backlog:0
mem_clients_slaves:0
mem_clients_normal:82008
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0
lazyfreed_objects:0

# Persistence
loading:0
current_cow_size:0
current_cow_size_age:0
current_fork_perc:0.00
current_save_keys_processed:0
current_save_keys_total:0
rdb_changes_since_last_save:0
rdb_bgsave_in_progress:0
rdb_last_save_time:1678726145
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0
module_fork_in_progress:0
module_fork_last_cow_size:0

# Stats
total_connections_received:4
total_commands_processed:18
instantaneous_ops_per_sec:0
total_net_input_bytes:1155
total_net_output_bytes:15317
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
expired_stale_perc:0.00
expired_time_cap_reached_count:0
expire_cycle_cpu_milliseconds:0
evicted_keys:0
keyspace_hits:904
keyspace_misses:0
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
total_forks:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
tracking_total_keys:0
tracking_total_items:0
tracking_total_prefixes:0
unexpected_error_replies:0
total_error_replies:0
dump_payload_sanitizations:0
total_reads_processed:17
total_writes_processed:16
io_threaded_reads_processed:0
io_threaded_writes_processed:0

# Replication
role:master
connected_slaves:0
master_failover_state:no-failover
master_replid:222db14d278334ec23b200a9047b533b424d3efc
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:0.062918
used_cpu_user:0.125837
used_cpu_sys_children:0.001571
used_cpu_user_children:0.002220
used_cpu_sys_main_thread:0.057831
used_cpu_user_main_thread:0.115663

# Modules
module:name=search,ver=20405,api=1,filters=0,usedby=[],using=[ReJSON],options=[handle-io-errors]
module:name=ReJSON,ver=999999,api=1,filters=0,usedby=[search],using=[],options=[handle-io-errors]

# Commandstats
cmdstat_hlen:calls=1,usec=4,usec_per_call=4.00,rejected_calls=0,failed_calls=0
cmdstat_hello:calls=3,usec=17,usec_per_call=5.67,rejected_calls=0,failed_calls=0
cmdstat_info:calls=4,usec=229,usec_per_call=57.25,rejected_calls=0,failed_calls=0
cmdstat_FT.INFO:calls=1,usec=56,usec_per_call=56.00,rejected_calls=0,failed_calls=0
cmdstat_client:calls=1,usec=17026,usec_per_call=17026.00,rejected_calls=0,failed_calls=0
cmdstat_type:calls=1,usec=3,usec_per_call=3.00,rejected_calls=0,failed_calls=0
cmdstat_ping:calls=3,usec=20,usec_per_call=6.67,rejected_calls=0,failed_calls=0
cmdstat_hscan:calls=1,usec=9,usec_per_call=9.00,rejected_calls=0,failed_calls=0
cmdstat_scan:calls=1,usec=165,usec_per_call=165.00,rejected_calls=0,failed_calls=0
cmdstat_config:calls=1,usec=35,usec_per_call=35.00,rejected_calls=0,failed_calls=0
cmdstat_ttl:calls=1,usec=2,usec_per_call=2.00,rejected_calls=0,failed_calls=0

# Errorstats

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=300,expires=0,avg_ttl=0

------ CLIENT LIST OUTPUT ------
id=11 addr=172.17.0.1:54130 laddr=172.17.0.2:6379 fd=8 name=127.0.0.1@6379 age=40 idle=10 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=12 addr=172.17.0.1:54132 laddr=172.17.0.2:6379 fd=9 name= age=15 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=668 qbuf-free=40286 argv-mem=534 obl=0 oll=0 omem=0 tot-mem=62166 events=r cmd=FT.SEARCH user=default redir=-1
id=13 addr=172.17.0.1:54134 laddr=172.17.0.2:6379 fd=10 name= age=15 idle=15 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=hello user=default redir=-1
id=14 addr=172.17.0.1:54136 laddr=172.17.0.2:6379 fd=11 name= age=8 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=40954 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=61456 events=r cmd=info user=default redir=-1

------ CURRENT CLIENT INFO ------
id=12 addr=172.17.0.1:54132 laddr=172.17.0.2:6379 fd=9 name= age=15 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=668 qbuf-free=40286 argv-mem=534 obl=0 oll=0 omem=0 tot-mem=62166 events=r cmd=FT.SEARCH user=default redir=-1
argv[0]: 'FT.SEARCH'
argv[1]: 'my_index'
argv[2]: '@number:[0 100]=>[KNN $K @my_vector $BLOB AS scores]'
argv[3]: 'LIMIT'
argv[4]: '0'
argv[5]: '5'
argv[6]: 'PARAMS'
argv[7]: '4'
argv[8]: 'K'
argv[9]: ''
argv[10]: 'BLOB'
argv[11]: '
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�
�)�'
argv[12]: 'DIALECT'
argv[13]: '2'
argv[14]: 'RETURN'
argv[15]: '2'
argv[16]: 'number'
argv[17]: 'scores'
argv[18]: 'SORTBY'
argv[19]: 'scores'
argv[20]: 'ASC'

------ MODULES INFO OUTPUT ------
# ReJSON_trace
ReJSON_trace:   0: redis_module::base_info_func
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-module-1.0.0/src/lib.rs:78:33
   1: rejson::__info_func
             at /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-module-1.0.0/src/macros.rs:120:13
   2: modulesCollectInfo
             at /usr/src/redis/src/module.c:7078:9
   3: logModulesInfo
             at /usr/src/redis/src/debug.c:1598:22
   4: printCrashReport
             at /usr/src/redis/src/debug.c:1849:5
      sigsegvHandler
             at /usr/src/redis/src/debug.c:1831:5
   5: <unknown>
   6: insertResultToHeap
             at /build/src/hybrid_reader.c:77:10
   7: computeDistances
             at /build/src/hybrid_reader.c:162:7
   8: HR_ReadHybridUnsorted
             at /build/src/hybrid_reader.c:227:5
   9: rpidxNext
             at /build/src/result_processor.c:89:10
  10: rpvecsimNext
             at /build/src/result_processor.c:259:8
  11: rpsortNext_innerLoop
             at /build/src/result_processor.c:403:12
      rpsortNext_Accum
             at /build/src/result_processor.c:494:16
  12: rploaderNext
             at /build/src/result_processor.c:677:12
  13: sendChunk
             at /build/src/aggregate/aggregate_exec.c:223:8
  14: AREQ_Execute
             at /build/src/aggregate/aggregate_exec.c:289:3
  15: execCommandCommon
             at /build/src/aggregate/aggregate_exec.c:418:5
  16: RedisModuleCommandDispatcher
             at /usr/src/redis/src/module.c:694:5
  17: call
             at /usr/src/redis/src/server.c:3721:5
  18: processCommand
             at /usr/src/redis/src/server.c:4241:9
  19: processCommandAndResetClient
             at /usr/src/redis/src/networking.c:2039:9
      processInputBuffer
             at /usr/src/redis/src/networking.c:2140:17
  20: callHandler
             at /usr/src/redis/src/connhelpers.h:79:18
      connSocketEventHandler
             at /usr/src/redis/src/connection.c:295:14
  21: aeProcessEvents
             at /usr/src/redis/src/ae.c:427:17
  22: aeMain
             at /usr/src/redis/src/ae.c:487:9
  23: main
             at /usr/src/redis/src/server.c:6401:5
  24: __libc_start_main
  25: _start


------ FAST MEMORY TEST ------
1:M 13 Mar 2023 16:49:56.046 # Bio thread for job type #0 terminated
1:M 13 Mar 2023 16:49:56.047 # Bio thread for job type #1 terminated
1:M 13 Mar 2023 16:49:56.047 # Bio thread for job type #2 terminated
*** Preparing to test memory region 55e997bb5000 (2277376 bytes)
*** Preparing to test memory region 55e998546000 (409600 bytes)
*** Preparing to test memory region 7fb9ce7d6000 (9437184 bytes)
*** Preparing to test memory region 7fb9d0000000 (135168 bytes)
*** Preparing to test memory region 7fb9d410f000 (3670016 bytes)
*** Preparing to test memory region 7fb9d5056000 (3145728 bytes)
*** Preparing to test memory region 7fb9d6800000 (8388608 bytes)
*** Preparing to test memory region 7fb9d7000000 (4194304 bytes)
*** Preparing to test memory region 7fb9d753e000 (2621440 bytes)
*** Preparing to test memory region 7fb9d77bf000 (8388608 bytes)
*** Preparing to test memory region 7fb9d7fc0000 (8388608 bytes)
*** Preparing to test memory region 7fb9d87c1000 (8388608 bytes)
*** Preparing to test memory region 7fb9d8fc2000 (8388608 bytes)
*** Preparing to test memory region 7fb9d97c3000 (8388608 bytes)
*** Preparing to test memory region 7fb9d9fc4000 (8388608 bytes)
*** Preparing to test memory region 7fb9dabf9000 (28672 bytes)
*** Preparing to test memory region 7fb9dac00000 (8388608 bytes)
*** Preparing to test memory region 7fb9db5c2000 (24576 bytes)
*** Preparing to test memory region 7fb9db789000 (16384 bytes)
*** Preparing to test memory region 7fb9db7ab000 (16384 bytes)
*** Preparing to test memory region 7fb9dba9f000 (16384 bytes)
*** Preparing to test memory region 7fb9dbc80000 (8192 bytes)
*** Preparing to test memory region 7fb9dbcb0000 (4096 bytes)
.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O
Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.

------ DUMPING CODE AROUND EIP ------
Symbol: (null) (base: (nil))
Module: /usr/lib/redis/modules/redisearch.so (base 0x7fb9da7c4000)
$ xxd -r -p /tmp/dump.hex /tmp/dump.bin
$ objdump --adjust-vma=(nil) -D -b binary -m i386:x86-64 /tmp/dump.bin
------

=== REDIS BUG REPORT END. Make sure to include from START to END. ===

       Please report the crash by opening an issue on github:

           http://github.com/redis/redis/issues

  Suspect RAM error? Use redis-server --test-memory to verify it.

```

**Additional information**

docker image
![image](https://user-images.githubusercontent.com/34903051/224772523-7bdcbf2f-a595-4d1c-8245-f25c7bec748d.png)

2. Steps to reproduce (if any)

```pom
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>2.4.5</version>
        <relativePath/> <!-- lookup parent from repository -->
    </parent>

     <dependency>
            <groupId>com.redis</groupId>
            <artifactId>spring-lettucemod</artifactId>
            <version>3.3.1</version>
        </dependency>
```

```java

    @Bean(
            name = {"objectRedisModulesConnection"},
            destroyMethod = "close"
    )
    StatefulRedisModulesConnection<String, Object> objectRedisModulesConnection(RedisModulesClient redisModulesClient) {
        return redisModulesClient.connect(new RedisCodec<String, Object>() {
            @Override
            public String decodeKey(ByteBuffer bytes) {
                bytes.position(0);
                byte[] keyBytes = new byte[bytes.remaining()];
                bytes.get(keyBytes);
                return new String(keyBytes);
            }

            @Override
            public Object decodeValue(ByteBuffer bytes) {
                if (bytes.remaining() == Integer.BYTES) { // int类型
                    return bytes.getInt();
                } else if (bytes.hasArray()) { // byte数组类型
                    return Arrays.copyOfRange(bytes.array(), bytes.position(), bytes.limit());
                } else { // 字符串类型
                    byte[] bytesArray = new byte[bytes.remaining()];
                    bytes.get(bytesArray);
                    return new String(bytesArray, StandardCharsets.UTF_8);
                }
            }


            @Override
            public ByteBuffer encodeKey(String key) {
                byte[] keyBytes = key.getBytes();
                return ByteBuffer.wrap(keyBytes);
            }

            @Override
            public ByteBuffer encodeValue(Object value) {
                if (value instanceof Integer) {
                    int intValue = (Integer) value;
                    byte[] bytes = ByteBuffer.allocate(4).putInt(intValue).array();
                    return ByteBuffer.wrap(bytes);
                } else if (value instanceof byte[]) {
                    byte[] byteArray = (byte[]) value;
                    return ByteBuffer.wrap(byteArray);
                } else if (value instanceof String) {
                    byte[] bytes = ((String) value).getBytes(StandardCharsets.UTF_8);
                    ByteBuffer buffer = ByteBuffer.allocate(bytes.length);
                    buffer.put(bytes);
                    buffer.flip();
                    return buffer;
                } else {
                    throw new IllegalArgumentException("Unsupported value type: " + value.getClass());
                }
            }
        });
    }

```

![image](https://user-images.githubusercontent.com/34903051/224772725-a96b542d-53f3-4a51-a3ef-41c5aff3aa82.png)

![image](https://user-images.githubusercontent.com/34903051/224773159-c3289c2a-05ce-42ed-bfe8-c4d5dbfc9243.png)

**Comment by GuyAv46** [comment]
Thanks for reporting!
It seems that we have a bug when searching for 0 vectors with a filter. I see that in the crash report the `K` value was empty and therefore we treated it as 0.
Did you use redispy client? and if so which version? it seems like the client failed to pass the `K` value correctly

**Comment by GuyAv46** [comment]
Solved on 2.8, 2.6.7, 2.4.17

**Comment by xiguazhiPrince** [comment]
> Thanks for reporting!
> It seems that we have a bug when searching for 0 vectors with a filter. I see that in the crash report the `K` value was empty and therefore we treated it as 0.
> Did you use Jedis client? and if so which version? it seems like the client failed to pass the `K` value correctly

I am using spring-lettucemod 3.3.1
I see that you have fixed the crashed bug of redis, so should I mention an issue in spring-lettucemod 's repository
